### PR TITLE
fixed for issue 15

### PR DESCRIPTION
**Pull request by francisuk1989**
see issue - https://github.com/TunSafe/TunSafe/issues/15

I have tested on Arch x86_64 using clang 7.0.1, Target: x86_64-pc-linux-gnu

git clone https://github.com/TunSafe/TunSafe.git
cd TunSafe
sudo pacman -S clang gcc7
make
sudo make install

### CODE REVIEW

**Comment by floating-cat** [comment]
This PR also works in Fedora (I tested).

**Comment by francisuk1989** [comment]
Heads up @floating-cat thanks for testing on Fedora :+1:

File: build_linux.sh [hunk 1]
```diff
@@ -5,8 +5,8 @@ RELARGS="-O3 -DNDEBUG"
 DBGARGS="-g -D_DEBUG"
 CURARGS="$RELARGS"
 
-clang++-6.0 -c -march=skylake-avx512 crypto/poly1305/poly1305-x64-linux.s crypto/chacha20/chacha20-x64-linux.s
-clang++-6.0 -I . $CURARGS -DWITH_NETWORK_BSD=1 -mssse3 -pthread -lrt -o tunsafe \
+clang++- -c -march=skylake-avx512 crypto/poly1305/poly1305-x64-linux.s crypto/chacha20/chacha20-x64-linux.s
```
Comment (floating-cat) [inline build_linux.sh:8]:
```suggestion
clang++ -c -march=skylake-avx512 crypto/poly1305/poly1305-x64-linux.s crypto/chacha20/chacha20-x64-linux.s
```

File: build_linux.sh [hunk 2]
```diff
@@ -5,8 +5,8 @@ RELARGS="-O3 -DNDEBUG"
...
+clang++- -I . $CURARGS -DWITH_NETWORK_BSD=1 -mssse3 -pthread -lrt -o tunsafe \
```
Comment (floating-cat) [inline build_linux.sh:9]:
```suggestion
clang++ -I . $CURARGS -DWITH_NETWORK_BSD=1 -mssse3 -pthread -lrt -o tunsafe \
```

**Comment by floating-cat** [comment]
I have realized you forget to remove `-` in your PR. My first comment is misleading (sorry), the `clang++` works in my Feodra but not the `clang++-`. I guess the same applies to your Arch?

**Comment by ghost** [comment]
why are you forcing use of clang... default in Fedora is gcc
cc -v
Using built-in specs.
COLLECT_GCC=/bin/cc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.3.1 20190223 (Red Hat 8.3.1-2) (GCC)
[root@localhost TunSafe]# cat build_linux.sh
#!/bin/sh
set -e

RELARGS="-O3 -DNDEBUG"
DBGARGS="-g -D_DEBUG"
CURARGS="$RELARGS"

c++ -c -march=skylake-avx512 crypto/poly1305/poly1305-x64-linux.s crypto/chacha20/chacha20-x64-linux.s
c++ -I . $CURARGS -DWITH_NETWORK_BSD=1 -mssse3 -pthread -lrt -o tunsafe 
tunsafe_amalgam.cpp 
crypto/aesgcm/aesni_gcm-x64-linux.s 
crypto/aesgcm/aesni-x64-linux.s 
crypto/aesgcm/ghash-x64-linux.s 
chacha20-x64-linux.o 
poly1305-x64-linux.o 
[root@localhost TunSafe]# gmake clean
gmake: *** No rule to make target 'clean'. Stop.
[root@localhost TunSafe]# make clean
make: *** No rule to make target 'clean'. Stop.
[root@localhost TunSafe]# make
sh ./build_linux.sh
ls -al tunsafe
-rwxr-xr-x. 1 root root 340728 Mar 28 12:09 tunsafe

works fine......

**Review by renierwoo — APPROVED** [review]

**Comment by francisuk1989** [comment]
closed.

**Pull request closed without merging**